code/game/machinery/doors/airlock.dm

```diff
@@ -22,24 +22,13 @@
 /// Someone, for the love of god, profile this.  Is there a reason to cache mutable_appearance
 /// if so, why are we JUST doing the airlocks when we can put this in mutable_appearance.dm for
 /// everything
-/proc/get_airlock_overlay(icon_state, icon_file, atom/offset_spokesman, em_block)
-	var/static/list/airlock_overlays = list()
-
-	var/base_icon_key = "[icon_state][REF(icon_file)]"
-	if(!(. = airlock_overlays[base_icon_key]))
-		. = airlock_overlays[base_icon_key] = mutable_appearance(icon_file, icon_state)
-	if(isnull(em_block))
-		return
-
-	var/turf/our_turf = get_turf(offset_spokesman)
-
-	var/em_block_key = "[base_icon_key][em_block][GET_TURF_PLANE_OFFSET(our_turf)]"
-	var/mutable_appearance/em_blocker = airlock_overlays[em_block_key]
-	if(!em_blocker)
-		em_blocker = airlock_overlays[em_block_key] = mutable_appearance(icon_file, icon_state, offset_spokesman = offset_spokesman, plane = EMISSIVE_PLANE, appearance_flags = EMISSIVE_APPEARANCE_FLAGS)
-		em_blocker.color = em_block ? GLOB.em_block_color : GLOB.emissive_color
-
-	return list(., em_blocker)
+/proc/get_airlock_overlay(icon_state, icon_file)
+	var/obj/machinery/door/airlock/A
+	pass(A)	//suppress unused warning
+	var/list/airlock_overlays = A.airlock_overlays
+	var/iconkey = "[icon_state][icon_file]"
+	if((!(. = airlock_overlays[iconkey])))
+		. = airlock_overlays[iconkey] = mutable_appearance(icon_file, icon_state)
```

Comment:
do not do assignment in an if statement (the old code was bad), also isn't this dropping functionality from our own implementation of this proc? what does it give us? is that handled by the overlays being mostly objects now?

Reply:
@LemonInTheDark on local copy i reverted unnecessary changes to this proc and fixed the if statement assignment

I still removed the em_block stuff because I don't know how to work that emissive stuff into this system [there is no emissive related airlock code on yogs so not sure]

Reply:
Ok we'll need to fix that, it's pretty important.
Does it actually impact the vis_contents stuff? the proc returns mutable appearances not objects

Reply:
emissives are back